client/src/components/Header/Header.js

```diff
@@ -1,44 +1,67 @@
-import React from "react";
+import React, { useState } from "react";
 import {
     Toolbar,
     AppBar,
     Button,
-    TextField
+    TextField,
+    Grid,
+    Box
 } from "@material-ui/core";
 import { NavLink } from "react-router-dom";
 import Logo from "../common/Logo/Logo";
 import { useStyles } from './styles';
 
 const Header = () => {
     const classes = useStyles();
+    const products = [];
+    const [seachInput, setSearchInput] = useState('');
+    const filteredProducts = () => {
+        const filteredProduct = products.filter(product => {
```

Comment:
Если продукт один, то лучше использовать find
```suggestion
        const filteredProduct = products.find(product => {
```

Если он может быть не один, то переменую нужно назвать иначе, не в единственном числе

Reply:
Это пока нет данных с сервера. после же нужно будет получать список продуктов, чтобы корректно работал поиск, продукт будет не один. В 16 строчне просто написан пустой масив как моковые данные

Reply:
Я понял, что пока данных нет, и что используется моковый массив, но тогда лучше назвать переменную **`filteredProducts`** или типа того. В рабочем коде массив с названием `filteredProduct`, содержащие _многие_ товары из сервера, будет смотреться странно

Reply:
fixed